src/figvalidationservice/src/test/java/ca/bc/gov/open/pssg/rsbc/dps/figvalidationservice/ValidateApplicantForSharingControllerTest.java

```diff
@@ -0,0 +1,76 @@
+package ca.bc.gov.open.pssg.rsbc.dps.figvalidationservice;
+
+import ca.bc.gov.open.ords.figcr.client.api.model.ValidateApplicantForSharingOrdsResponse;
+import ca.bc.gov.open.pssg.rsbc.dps.figvalidationservice.exception.FigaroValidationServiceException;
+import ca.bc.gov.open.pssg.rsbc.dps.figvalidationservice.types.ValidateApplicantForSharingResponse;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class ValidateApplicantForSharingControllerTest {
+
+
+    public static final String VALID_STATUS_CODE = "0";
+    public static final String INVALID_STATUS_CODE = "-2";
+    public static final String VALID = "valid";
+    public static final String INVALID = "invalid";
+    public static final String VALIDATION_RESULT = "validationresult";
+
+    private ValidateApplicantForSharingController sut;
+
+    @Mock
+    private FigaroValidation figaroValidationMock;
+
+    @BeforeAll
+    public void init() throws FigaroValidationServiceException {
+
+        ValidateApplicantForSharingOrdsResponse validResponse =
+                new ValidateApplicantForSharingOrdsResponse().statusCode(VALID_STATUS_CODE).statusMessage(VALID).validationResult(VALIDATION_RESULT);
+        ValidateApplicantForSharingOrdsResponse erroResponse =
+                new ValidateApplicantForSharingOrdsResponse().statusCode(INVALID_STATUS_CODE).statusMessage(INVALID);
+
+        MockitoAnnotations.initMocks(this);
+
+        Mockito.doReturn(validResponse).when(figaroValidationMock).validateApplicantForSharing(ArgumentMatchers.argThat(x -> x.getApplPartyId().equals("1")));
+        Mockito.doReturn(erroResponse).when(figaroValidationMock).validateApplicantForSharing(ArgumentMatchers.argThat(x -> x.getApplPartyId().equals("2")));
+        Mockito.doThrow(FigaroValidationServiceException.class).when(figaroValidationMock).validateApplicantForSharing(ArgumentMatchers.argThat(x -> x.getApplPartyId().equals("EXCEPTION")));
+
+        sut = new ValidateApplicantForSharingController(figaroValidationMock);
+
+    }
+
+
+    @Test
+    public void withValidParametersShouldReturnValidResponse() {
+
+        ValidateApplicantForSharingResponse response = sut.validateApplicantForSharing("1", "TEST");
+        Assertions.assertEquals(0, response.getRespCode());
```

Comment:
Maybe use the constants from above for the response codes (2 places)?

Reply:
Yes right, thanks

Reply:
changed

Reply:
this needs to be an integer, the other one is a string